dask/array/core.py

```diff
@@ -2702,65 +2711,37 @@ def insert_to_ooc(arr, out, lock=True, region=None, return_stored=False):
         lock = Lock()
 
     slices = slices_from_chunks(arr.chunks)
+    if region:
+        slices = [fuse_slice(region, slc) for slc in slices]
 
     name = 'store-%s' % arr.name
+    sto_chunk = store_chunk
+    args = ()
+    if return_stored and load_stored:
+        name = 'load-store-%s' % arr.name
+        sto_chunk = load_store_chunk
+        args = (load_stored,)
+
     dsk = dict()
     for t, slc in zip(core.flatten(arr.__dask_keys__()), slices):
         store_key = (name,) + t[1:]
-        dsk[store_key] = (store_chunk, t, out, slc, lock, region, return_stored)
+        dsk[store_key] = (sto_chunk, t, out, slc, lock, return_stored) + args
```

Comment:
nit: I'd write this as a dictionary comprehension, but that's just my opinion. No strong feelings.

Reply:
Felt like I had made a lot of changes in the code and was trying to keep it easy to review. So didn't rewrite this, but sure we can give it a try.